main.go

```diff
@@ -98,6 +139,39 @@ func StartServer(cfg *WebserverConfig, mux http.Handler) error {
 		MaxHeaderBytes: 1 << 20, // 1mb
 	}
 
+	if len(cfg.AutocertDomains) != 0 {
+		log.Printf("Starting server on ports 443 and 80, using autocert domains %v.", cfg.AutocertDomains)
+
+		cache := autocert.DirCache(cfg.AutocertDir)
+		manager := &autocert.Manager{
+			Cache:      cache,
+			Prompt:     autocert.AcceptTOS,
+			HostPolicy: autocert.HostWhitelist(cfg.AutocertDomains...),
+		}
+
+		certHandler := manager.HTTPHandler(http.HandlerFunc(redirectToHTTPS))
```

Comment:
Interesting! I didn't know about `tls-alpn-01`. Thanks! I'll update the PR.

One question regarding port 80. It also serves as redirect from http to https. Is it needed for LN address functionality? Is it worth having it under an option? So nginx is not needed to do this redirect.

Reply:
Sorry, it's been a busy week :sweat_smile:

The LNURL spec says URLs can either be `https://`, or they can be `http://` if the domain name ends with `.onion` ([source](https://github.com/lnurl/luds/blob/luds/01.md#https-or-onion)), so i think we can get away without HTTP here. Usually HTTP -> HTTPS redirection is set up for compatibility with old web browsers, which still default to plaintext HTTP connections when you type a bare domain into the browser URL bar. Wallets on the other hand should _never_ be making plaintext requests when asking for lightning invoices, because a middle-man could serve fake responses.

If a user needs an HTTP->HTTPS redirection server for some reason, it's very easy to write a second program or script which handles that redirect independently. Most popular webservers like nginx allow you to configure HTTP -> HTTPS redirectors very easily too.

In the interest of avoiding clutter and complexity in the config, we should keep things simple and just expose a single port, which could either be listening with TLS (via autocert), or listening on a plaintext TCP socket (while some upstream program terminates TLS for us).